internal/versiongen/gen_test.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/go-version"
+)
+
+func TestGetTerraformReleases(t *testing.T) {
+	releases, err := GetTerraformReleases()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expectedLength := 234
+	if expectedLength < len(releases) {
+		t.Fatalf("expected >= %d releases, %d given", expectedLength, len(releases))
+	}
+
+	expectedDate := time.Date(2017, 3, 1, 17, 36, 49, 0, time.UTC)
+	expectedOldestRelease := release{
+		Version: version.Must(version.NewVersion("0.6.4")),
```

Comment:
Why is this the oldest release? Is it because the API returns this as the last entry of the array currently?
Should we sort the releases first to avoid a potential flaky test?

Reply:
I was asking myself the same question haha!

The dates for oldest releases were backfilled from an old S3 bucket which happened to be migrated and the original dates from the old bucket were not retained at the time.